### ISSUE TITLE
[docs] The doc says that community pickers have nested import on the pro package, which is false

### ISSUE THREAD

**Issue by chrismowbray** [mui/mui-x#8885]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Below are 3 examples imports are that the docs say are supported but in reality, they are not.

[MobileTimePicker Docs](https://mui.com/x/api/date-pickers/mobile-time-picker/)
[MobileDatePicker Docs](https://mui.com/x/api/date-pickers/mobile-date-picker/#import)
[LocalizationProvider Docs](https://mui.com/x/react-date-pickers/getting-started/)

```tsx
import { MobileTimePicker } from "@mui/x-date-pickers-pro/MobileTimePicker";
import { MobileDatePicker } from "@mui/x-date-pickers-pro/MobileDatePicker";
import { LocalizationProvider } from "@mui/x-date-pickers-pro/LocalizationProvider";
```

Im not sure if the docs needs updated, or the functionality needs added. 

### Current behavior 😯

Have to use barrel import "@mui/x-date-pickers-pro" and thus increasing the bundle sizea

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by alexfauquette** [comment]
It's a docs issue. The prop package reexports the MIT components but does not have the file structure.

So `import { MobileTimePicker } from '@mui/x-date-pickers-pro/MobileTimePicker';` can not be done

See this [docs section](https://mui.com/x/react-date-pickers/base-concepts/#imports) for more details

**Comment by oliviertassinari** [comment]
> Have to use barrel import "@mui/x-date-pickers-pro" and thus increasing the bundle sizea

@chrismowbray This is tree-shaked in production. But yeah, in dev mode, it makes things a bit slower.

**Comment by alexfauquette** [comment]
Fixed by #9898
